logstash-core/lib/logstash/agent.rb

```diff
@@ -260,16 +264,17 @@ def create_pipeline(settings, config=nil)
     end
   end
 
-  def fetch_config(settings)
-    @config_loader.format_config(settings.get("path.config"), settings.get("config.string"))
+  def fetch_config
+    # TODO(ph) multiple pipeline, we only support one config for now
+    @source_loader.fetch.first
```

Comment:
when other pipelines are loaded (like the monitoring pipeline), we aren't sure which one is loaded first.

This single pipeline strategy should be avoided and become something like:

```ruby
def reload_state!
   pipeline_configs = @source_loader.fetch
   pipelines_to_create, pipelines_to_delete, pipelines_to_reload = resolve_state_actions(pipeline_configs)
   pipelines_to_create.each {|p| create_pipeline(p) }
   pipelines_to_delete.each {|p| delete_pipepline(p) }
   pipelines_to_reload.each {|p| reload_pipeline(p) }
end
```